### ISSUE TITLE
MongoDB oauth User model

### ISSUE THREAD

**Issue by egelados** [fastapi-users/fastapi-users#904]
**TL:DR**

> **class User(models.BaseUser)**

should be replaced by
> **class User(models.BaseUser, models.BaseOAuthAccountMixin)**

in models.py for the MongoDB oauth to work properly

### Description

The oauth full example (as well as the rest of the documentation) mentions the basic models configuration is adequate for the mongoDB implementation of oauth.

However regarding the User model, in case we do not additionally inherit from the **BaseOAuthAccountMixin**, which adds a List of **BaseOAuthAccount** objects, we do not provide for the User a way to store the OAuth information (access tokens, account ids...)

So it seems only reasonable that should we use the code displayed on the https://fastapi-users.github.io/fastapi-users/configuration/oauth/ quoting:

> class User(models.BaseUser):
>     pass
> 

it registers the User document in our collection but it does not store the **oauth_accounts** array in it that should look like that:

>   "oauth_accounts": [
>     {
>       "id": {
>         "$binary": {
>           "base64": "#####################==",
>           "subType": "04"
>         }
>       },
>       "oauth_name": "google",
>       "access_token": "############################################################",
>       "expires_at": 1643051360,
>       "refresh_token": null,
>       "account_id": "people/######################",
>       "account_email": "###########@gmail.com"
>     }
>   ]
> 

Hence in case we try to access our API endpoints bearing this user's token it fails with a relevant error message quoting the lack of the **oauth_accounts** part of the document.

### Solution

To overcome this problem i used a User model similar to the Tortoise ORM full example inheriting from both the  **BaseUser** AND the **BaseOAuthAccountMixin**:

> class User(models.BaseUser, models.BaseOAuthAccountMixin):
>     pass
>

**Comment by frankie567** [comment]
There was indeed a mistake in the MongoDB example for OAuth, thank you for raising this 🙏

**Comment by egelados** [comment]
Thank you for your great project. 
It would be fantastic if you considered incorporating some kind of permissions system!
